packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -148,38 +161,73 @@ export class GreasedLinePluginMaterial extends MaterialPluginBase {
+    private static _EmptyColorsTexture: BaseTexture;
+
     private _colorsTexture?: RawTexture;
 
-    private _dashArray = 0;
+    private _visibility: number;
+    private _useDash: boolean;
+    private _dashCount: number;
+    private _dashRatio: number;
+    private _dashOffset: number;
+    private _dashArray: number;
+    private _width: number;
+    private _sizeAttenuation: boolean;
+    private _color: Color3 | undefined;
+    private _colorMode: GreasedLineMeshColorMode;
+    private _colors: Nullable<Color3[]> | undefined;
+    private _useColors: boolean;
+    private _colorDistributionType: GreasedLineMeshColorDistributionType;
+    private _colorsSampling: number;
+    private _resolution: Vector2;
 
     private _options: GreasedLineMaterialOptions;
 
     private _engine: Engine;
 
-    private static _EmptyColorsTexture: BaseTexture;
-
-    constructor(material: Material, private _scene: Scene, options: GreasedLineMaterialOptions) {
-        options = options ?? {};
+    constructor(material: Material, private _scene: Scene, options?: GreasedLineMaterialOptions) {
+        options = options || {
+            color: GreasedLinePluginMaterial.DEFAULT_COLOR,
+        };
 
         const defines = new MaterialGreasedLineDefines();
-        defines.GREASED_LINE_HAS_COLOR = !!options.color;
+        defines.GREASED_LINE_HAS_COLOR = !!options.color ?? false;
         defines.GREASED_LINE_SIZE_ATTENUATION = options.sizeAttenuation ?? false;
         defines.GREASED_LINE_COLOR_DISTRIBUTION_TYPE_LINE = options.colorDistributionType === GreasedLineMeshColorDistributionType.COLOR_DISTRIBUTION_TYPE_LINE;
-
         super(material, GreasedLinePluginMaterial.GREASED_LINE_MATERIAL_NAME, 200, defines);
 
         this._options = options;
+
         this._scene = this._scene ?? material.getScene();
         this._engine = this._scene.getEngine();
 
-        if (options.colors) {
-            this._createColorsTexture(`${material.name}-colors-texture`, options.colors);
+        this._visibility = options.visibility ?? 1;
+        this._useDash = options.useDash ?? false;
+        this._dashRatio = options.dashRatio ?? 0.5;
+        this._dashOffset = options.dashOffset ?? 0;
+        this._width = options.width ? options.width : this._sizeAttenuation ? 1 : 0.1;
+        this._sizeAttenuation = options.sizeAttenuation ?? false;
+        this._colorMode = options.colorMode ?? GreasedLineMeshColorMode.COLOR_MODE_SET;
+        this._color = options.color;
+        this._useColors = options.useColors ?? false;
+        this._colorDistributionType = options.colorDistributionType ?? GreasedLineMeshColorDistributionType.COLOR_DISTRIBUTION_TYPE_SEGMENT;
+        this._colorsSampling = options.colorsSampling ?? RawTexture.NEAREST_NEAREST;
+        this._resolution = options.resolution ?? new Vector2(this._engine.getRenderWidth(), this._engine.getRenderHeight());
+        this._colors = options.colors; // call the setter, to create the color texture

```

Comment:
If you want to call the setter, you should use `this.colors` instead.

Reply:
Wrong comment. Removed.